src/plugins/intel_cpu/src/nodes/rdft.cpp

```diff
@@ -0,0 +1,1537 @@
+    for (auto& axis : axes) {
+        if (axis < 0) {
+            axis += rank;
+        }
+    }
+    if (!inverse)
+        std::sort(axes.begin(), axes.end(), std::greater<int>());
+    else
+        std::sort(axes.begin(), axes.end());
+}
+
+static std::vector<int> generate_fft_indices(int vlen);
+
+RDFT::RDFT(const std::shared_ptr<ngraph::Node>& op, const dnnl::engine& eng, WeightsSharing::Ptr &cache) :
+               Node(op, eng, cache) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        IE_THROW(NotImplemented) << errorMessage;
+    }
+
+    std::string error_msg_prefix = "RDFT layer with name '" + op->get_name() + "'";
+    const size_t num_inputs = getOriginalInputsNumber();
+    if (num_inputs != 2 && num_inputs != 3) {
+        IE_THROW() << error_msg_prefix << " has invalid number of input/output edges: " << num_inputs;
+    }
+
+    const auto axes_rank = inputShapes[AXES_INDEX].getRank();
+    if (axes_rank != 1) {
+        IE_THROW() << error_msg_prefix << " has invalid 'axes' input tensor with rank: " << axes_rank;
+    }
+
+    if (num_inputs > SIGNAL_SIZE_INDEX) {
+        const auto signal_size_rank = inputShapes[SIGNAL_SIZE_INDEX].getRank();
+        if (signal_size_rank != 1) {
+            IE_THROW() << error_msg_prefix << " has invalid 'signal_size' input tensor with rank: " << signal_size_rank;
+        }
+    }
+
+    inverse = ov::is_type<ov::op::v9::IRDFT>(op);
+}
+
+void RDFT::getSupportedDescriptors() {}
+
+void RDFT::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    const auto& data_precision = getOriginalInputPrecisionAtPort(DATA_INDEX);
+    if (!data_precision.is_float()) {
+        IE_THROW() << error_msg_prefix << " has unsupported 'data' input precision: " << data_precision.name();
+    }
+
+    const auto& axes_precision = getOriginalInputPrecisionAtPort(AXES_INDEX);
+    if (axes_precision != Precision::I32 && axes_precision != Precision::I64) {
+        IE_THROW() << error_msg_prefix << " has unsupported 'axes' input precision: " << axes_precision.name();
+    }
+
+    if (inputShapes.size() > SIGNAL_SIZE_INDEX) {
+        const auto& signal_size_precision = getOriginalInputPrecisionAtPort(SIGNAL_SIZE_INDEX);
+        if (signal_size_precision != Precision::I32 && signal_size_precision != Precision::I64) {
+            IE_THROW() << error_msg_prefix << " has unsupported 'signal_size' input precision: " << signal_size_precision.name();
+        }
+    }
+
+    std::vector<PortConfigurator> configurators({{LayoutType::ncsp, Precision::FP32},
+                                                 {LayoutType::ncsp, Precision::I32}});
+    if (inputShapes.size() > SIGNAL_SIZE_INDEX)
+        configurators.push_back({LayoutType::ncsp, Precision::I32});
+
+    addSupportedPrimDesc(configurators, {{LayoutType::ncsp, Precision::FP32}}, impl_desc_type::ref_any);
+}
+
+void RDFT::execute(dnnl::stream strm) {
```

Comment:
Do you use some known algorithm for RDFT here? As I understand, imaginary part is filled by zero and Cooley Tukey FFT approach is implemented here?

Reply:
> As I understand, imaginary part is filled by zero and Cooley Tukey FFT approach is implemented here?
right, FFT is based on Cooley Tukey and the imag part is filled with zeros. For non power-of-two cases, naive DFT is implemented and it works directly on real data.